include/linux/page.h

```diff
@@ -12,6 +13,11 @@ struct page;
 
 #endif
 
+#ifndef PAGE_SHIFT
+#define PAGE_SHIFT sysconf(_SC_PAGESIZE)
```

Comment:
Instead of this, let's just define PAGE_SHIFT to 12. In userspace there shouldn't be any need for PAGE_SIZE to match the hardware page size for what we're using it, and we do want it to be a constant - we'd want to check what sysconf(_SC_PAGESIZE) compiles to.